### PR TITLE
#8227 WFS  identify in Cesium and layer settings

### PR DESCRIPTION
**Pull request by barema4**
## Description
* This Pull Request allows one to display feature info of WFS layer in Cesium . It also allows the identity settings to add 
possibility to control the Identify type  for WFS layers in 2D and 3D  from layer settings. Allowing only the 
DISABLE IDENTIFY
PROPERTIES
TEMPLATE 

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [ ] Bugfix
 - [x] Feature
 - [ ] Code style update (formatting, local variables)
 - [ ] Refactoring (no functional changes, no api changes)
 - [ ] Build related changes
 - [ ] CI related changes
 - [ ] Other... Please describe:

## Issue

**What is the current behavior?**

* Does not add support for Get feature info requests for wfs layers in cesium
*  Does not make Identify settings available for this type of layer both 2d/3d
* #8227

**What is the new behavior?**

 * Adds support for Get feature info requests for wfs layers in cesium
 * Makes Identify settings available for this type of layer both 2d/3d

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

### CODE REVIEW

File: web/client/components/TOC/fragments/settings/FeatureInfo.jsx [hunk 1]
```diff
@@ -84,12 +84,15 @@ export default class extends React.Component {
      * @return {object} info formats
      */
     supportedInfoFormats = () => {
+        const excludedFormatsWfs = ['TEXT', 'HTML'];
         const availableInfoFormats =  this.props.element?.infoFormats || [];
+        const supportedWfsFormats = Object.fromEntries(Object.entries(this.props.defaultInfoFormat).filter(([key]) => !excludedFormatsWfs.includes(key)));
+        const formats = this.props.element.type === 'wfs' ? supportedWfsFormats : this.props.defaultInfoFormat;
```
Comment (MV88) [inline web/client/components/TOC/fragments/settings/FeatureInfo.jsx:90]: can you add a unit tests that will cover the list of info formats used by "wfs" and maybe another one like "wms" ?

you can render this comp and check the items rendered in the list

**Review by MV88 — CHANGES_REQUESTED** [review]
@barema4 hi
I noticed that feature info does not work nice if you click near the line in 2d mode (openlayers)
image

while if you click outside it works better
image

this can be noticed better in 3d since the circle drawn in the first case is not matching intersecting the line and in the second case is

I think there is something wrong in the way circle is created in 2d mode or on the point used?
try to create a vector GeoJson layer from the geometry created and used in the cql filter intersect, and then import it in the map just after having clicked somewhere.

I can show you what I mean

**Comment by barema4** [comment]
Done adding unit tests for supported info formats for wfs and wms layers

File: web/client/components/TOC/fragments/settings/__tests__/FeatureInfo-test.jsx [hunk 2]
```diff
@@ -83,4 +83,15 @@ describe("test FeatureInfo", () => {
         const testComponent = document.getElementsByClassName('test-preview');
         expect(testComponent.length).toBe(2);
     });
+    it('test rendering supported infoFormats for wfs layer', () => {
+        ReactDOM.render(<FeatureInfo element={{infoFormats: ["text/html", "application/json"]}} formatCards={formatCards} defaultInfoFormat={defaultInfoFormat} />, document.getElementById("container"));
+        const testComponent = document.getElementsByClassName('test-preview');
+        expect(testComponent.length).toBe(3);
```
Comment (MV88) [inline web/client/components/TOC/fragments/settings/__tests__/FeatureInfo-test.jsx:89]: can you check also the values inside?

File: web/client/components/TOC/fragments/settings/__tests__/FeatureInfo-test.jsx [hunk 3]
```diff
@@ -83,4 +83,15 @@ describe("test FeatureInfo", () => {
...
+    });
+
+    it('test rendering supported infoFormats for wms layer', () => {
+        ReactDOM.render(<FeatureInfo element={{infoFormats: ["text/html", "text/plain", "application/json"]}} formatCards={formatCards} defaultInfoFormat={defaultInfoFormat} />, document.getElementById("container"));
+        const testComponent = document.getElementsByClassName('test-preview');
+        expect(testComponent.length).toBe(4);
```
Comment (MV88) [inline web/client/components/TOC/fragments/settings/__tests__/FeatureInfo-test.jsx:95]: can you check also the values inside?

**Review by MV88 — CHANGES_REQUESTED** [review]

File: web/client/components/TOC/fragments/settings/__tests__/FeatureInfo-test.jsx (hunk 2 above)
Comment (barema4) [reply web/client/components/TOC/fragments/settings/__tests__/FeatureInfo-test.jsx]: Done checking for values inside

**Review by MV88 — CHANGES_REQUESTED** [review]
Hi @barema4  I checked your tests  for PR of info format and WFS, but these were not exactly testing what you have added in the codebase

Since the short time we have, I have already fixed them [here ](https://github.com/MV88/MapStore2/commit/2a478c24c6f32bae53416e7036bcdc95a31f7a02) 
can you include them in the PR?

if you like we can discuss about them, but the key concept is to recreate the runtime condition, i.e. passing right props to emulate a behaviour of the application, for example for the test for wfs, you missed to pass the type: "wfs" that was needed to test this component when a layer wfs is in use

please pay more attention to these details

**Comment by barema4** [comment]
Hi @MV88 , I have got  it  right. Thanks for calcification.  I have added them.

**Review by MV88 — APPROVED** [review]

**Comment by MV88** [comment]
@ElenaGallo please test in DEV

**Comment by tdipisa** [comment]
@ElenaGallo let us know if we can backport to 2022.01.xx-c040 branch

**Comment by ElenaGallo** [comment]
Test passed, @MV88 please backport to 2022.01.xx-c040 branch

**Pull request merged**